### PR TITLE
Enhancement watcher as manager

### PR DESCRIPTION
**Pull request by DSWallach**
# Pull Request Template for MMEDS

## What has changed
A lot 

### New Stuff:
- host/manager.py: This is the file to execute to start the Watcher running on the local network

### Changed Stuff:
- mmeds/database.py: Was split into three separate files: 
- - mmeds/database/database.py
- - mmeds/database/metadata_uploader.py
- - mmeds/database/sql_builder.py
These correspond to the three classes that used to all be contained with mmeds/database.py
- mmeds/config.py: Bunch of globals needed to be updated to play nice with the new setup
- mmeds/html/*.html: Added pages for the analysis query functionality. Probably should've been a separate pull request but here we are.
- mmeds/server.py: Many modifications to have it connect to the new watcher setup.
- mmeds/spawn.py: Modified to be a Manager object rather than a typical Python process.
- www_files/app.py: This is the WSGI app file that gets picked up by the apache server.
- www_files/htaccess: The `.htaccess` file for configuring the directory the WSGI app lives in.
## Checklist of pre-requisites
-   [x] Does the code run?  
-   [x] Does the code follow the repository style?  
-   [x] Is the code tested?  

## How to use the feature
Example: To do X run new function Y with parameters A, B, and C which gives result Z.

## How the design has changed
Previously the server was created by cherrypy as part of starting the application. MMEDs now requires a separate apache server to pick up and host it's application file.

### CODE REVIEW

File: mmeds/config.py [hunk 1]
```diff
@@ -11,39 +11,39 @@
...
     HTML_DIR = Path(html.__file__).parent.resolve()
-    CSS_DIR = Path(css.__file__).parent.resolve()
     STORAGE_DIR = Path(resources.__file__).parent.resolve()
     DATABASE_DIR = Path().home() / 'mmeds_server_data'
-    SESSION_PATH = DATABASE_DIR / '/CherryPySession'
+    SESSION_PATH = DATABASE_DIR / 'CherryPySession'
     SERVER_PATH = 'http://localhost/myapp/'
+    CSS_DIR = 'http://localhost/CSS/'
     IMAGE_PATH = str(CSS_DIR) + '/'
+    # SERVER_PATH = '{}:{}/'.format(sec.SERVER_HOST, sec.SERVER_PORT)
```
Comment (mstapylton) [inline mmeds/config.py:27]: commented out line

File: mmeds/config.py [hunk 2]
```diff
@@ -82,7 +82,7 @@
     'request.scheme': 'https',
     'Secureheaders.on': True,
     'tools.sessions.secure': True,
-    #'tools.sessions.httponly': True,
```
Comment (mstapylton) [inline mmeds/config.py:85]: commented out lines should probably be removed

File: mmeds/config.py [hunk 3]
```diff
@@ -214,7 +222,10 @@
 MODULE_ROOT = DATABASE_DIR.parent / '.modules/modulefiles'
 
 if not DATABASE_DIR.exists():
-    DATABASE_DIR.mkdir()
+    try:
+        DATABASE_DIR.mkdir()
+    except FileExistsError:
+        pass
```
Comment (mstapylton) [inline mmeds/config.py:228]: should at least log the exception I would think

File: mmeds/config.py [hunk 4]
```diff
@@ -253,10 +267,16 @@
 TEST_PATH = DATABASE_DIR / 'test_files'
 TEST_DIR = DATABASE_DIR / 'mmeds_test_dir'
 if not TEST_DIR.exists():
-    TEST_DIR.mkdir()
+    try:
+        TEST_DIR.mkdir()
+    except FileExistsError:
+        pass
```
Comment (mstapylton) [inline mmeds/config.py:273]: and log errror here

File: mmeds/config.py [hunk 5]
```diff
@@ -253,10 +267,16 @@
...
 TEST_DIR_0 = DATABASE_DIR / 'mmeds_test_dir0'
 if not TEST_DIR_0.exists():
-    TEST_DIR_0.mkdir()
+    try:
+        TEST_DIR_0.mkdir()
+    except FileExistsError:
+        pass
```
Comment (mstapylton) [inline mmeds/config.py:279]: and log error here

File: www_files/app.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+print("forcing reload")
+logger = MMEDSLog('wsgi-debug').logger
+
+from sys import argv
+
+import cherrypy as cp
+curdir = os.path.abspath(os.path.dirname(__file__))
+
+
```
Comment (mstapylton) [inline www_files/app.py:39]: two blank lines

File: www_files/app.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+# Using a global to prevent the app from being generated multiple times
+loaded = False
+
+print('saaaa')
```
Comment (mstapylton) [inline www_files/app.py:43]: saaaa?

**Review by mstapylton — CHANGES_REQUESTED** [review]
Mostly good, a few style changes suggested. Didn't look too closely at the files where you're just moving code around i.e. database.py.

File: mmeds/config.py (hunk 3 above)
Comment (DSWallach) [reply mmeds/config.py:228]: I don't think it's necessary. It's just for creating those directories the first time that the config is loaded on a particular machine. Logging that those files exist for every subsequent loading of this file is going to clog up the logs with a lot of unhelpful information

Comment (mstapylton) [reply mmeds/config.py:228]: fair enough

**Pull request merged**